pkg/apis/batch/validation/validation.go

```diff
@@ -546,6 +546,43 @@ func ValidateJobTemplateSpec(spec *batch.JobTemplateSpec, fldPath *field.Path, o
 	return allErrs
 }
 
+func validateCompletions(spec, oldSpec batch.JobSpec, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	isIndexedJob := spec.CompletionMode != nil && *spec.CompletionMode == batch.IndexedCompletion
+	atLeastOneCompletionsNil := spec.Completions == nil || oldSpec.Completions == nil
+	bothCompletionsNil := spec.Completions == nil && oldSpec.Completions == nil
+
+	// Non-indexed Jobs cannot have completions set at all.
+	if !isIndexedJob {
+		if !bothCompletionsNil {
```

Comment:
ValidateImmutableField should handle the case where both values are nil, shouldn't need this qualifier

Reply:
I see now, ok thanks - updated.